src/transformers/generation_utils.py

```diff
@@ -764,9 +1035,30 @@ def greedy_search(
             model_inputs = self.prepare_inputs_for_generation(input_ids, **model_kwargs)
 
             # forward pass to get next token
-            outputs = self(**model_inputs, return_dict=True)
+            outputs = self(
+                **model_inputs,
+                return_dict=True,
+                output_attentions=output_attentions,
+                output_hidden_states=output_hidden_states,
+            )
             next_token_logits = outputs.logits[:, -1, :]
 
+            # Store logits, attentions and hidden_states when required
+            if return_dict_in_generate:
+                if output_scores:
+                    logits += (next_token_logits,)
```

Comment:
Sorry I wasn't super clear on this, here I think it's actually better to add the `scores` from a couple of lines below. We should still call it logits though since it's just a processed version of the logits.

Reply:
Or maybe calling `scores` is actually better. This way we can consistently call the output `scores` for all other `GenerateOutputs` as well.